src/main/java/com/fulcrumgenomics/util/PickLongIndices.scala

```diff
@@ -0,0 +1,291 @@
+      i += 1
+      j -= 1
+    }
+    palindrome
+  }
+
+  /** If we have structure prediction parameters, returns Some(min deltaG) else None. */
+  private[util] def worstDeltaG(index: Index): Option[Double] = {
+    val sIndex = new String(index)
+    dnaFoldPredictor.map(folder => adapters.map(a => folder.predict(a.replaceAll("N+", sIndex)).deltaG()).min)
+  }
+
+  /** Generates a random index sequence of the desired length. */
+  private def randomIndex(length: Int): Index = {
+    val index = new Array[Byte](length)
+    forloop (0, length) { i => index(i) = Bases(random.nextInt(4)) }
+    index
+  }
+
+  /** Checks to see if the index sequence contains any homopolymers longer than longest. */
+  private[util] def homopolymerLengthOk(index: Index, longest: Int): Boolean = {
+    var end = index.length - longest
+    var ok = true
+
+    // Check for a homopolymer starting at each base up to longest+1 from the end of the index
+    forloop (0)(ok && _ < end)(_ + 1) { i =>
+      val base = index(i)
+      var j = i+1
+      var inHomoPolymer = true
+      forloop (i+1)(inHomoPolymer && _ < i+longest+1)(_ + 1) { j =>
+        if (base != index(j)) inHomoPolymer = false
+      }
+
+      ok = !inHomoPolymer
+    }
+
+    ok
+  }
+
+  /** Counts mismatches in two arrays of equal length, all upper-case non-ambiguous, bases. */
+  private[util] def mismatches(lhs: Index, rhs: Index): Int = {
+    var n = 0
+    forloop (0, lhs.length) { i => if (lhs(i) != rhs(i)) n += 1 }
+    n
+  }
+
+  /** Writes information about each index to a file. */
+  private def writeOutput(indices: IndexSet, output: FilePath): Unit = {
+    // Go through and compute pair-wise distance between all the indices
+    case class AnnotatedIndex(index: Index, existing: Boolean, distances: NumericCounter[Int] = new NumericCounter[Int])
+    val annotated = indices.map(index => AnnotatedIndex(index, existing=ExistingIndices.contains(index)))
+      .toList.sortBy(a => (!a.existing, new String(a.index)))
+
+    annotated.tails.foreach {
+      case x :: ys => ys.foreach { y =>
+        val n = mismatches(x.index, y.index)
+        x.distances.count(n)
+        y.distances.count(n)
+      }
+      case _ => Unit
+    }
+
+    val Sep = "\t"
+    val lines = ListBuffer[String]()
+    val dfmt = new DecimalFormat("0.000")
+
+    lines += Seq("barcode", "source", "min_mismatches", "barcodes_at_min_mismatches", "gc", "min_deltag").mkString(Sep)
+    lines ++= annotated.map(ann => Seq(
+      new String(ann.index),
+      if (ann.existing) "existing" else "novel",
+      ann.distances.headOption.map { case (distance, count) => distance }.getOrElse(ann.index.length),
+      ann.distances.headOption.map { case (distance, count) => count    }.getOrElse(0),
+      dfmt.format(SequenceUtil.calculateGc(ann.index)),
+      worstDeltaG(ann.index).getOrElse("n/a")
+    ).mkString(Sep))
+
+    Io.writeLines(output, lines)
+  }
```

Comment:
If you use a `Metric` class, other folks can read the output of this tool easily!